streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -345,7 +359,11 @@ void handleLostAll() {
                 cleanupTask(task);
                 task.closeDirty();
                 iterator.remove();
-                taskProducers.remove(task.id());
+                try {
+                    activeTaskCreator.closeAndRemoveTaskProducerIfNeeded(task.id());
+                } catch (final RuntimeException e) {
+                    log.debug("Error handling lostAll", e);
```

Comment:
1. Let's make it a warn instead of a debug.
2. The error message can be more specific here: `Error closing task producer for task {} while handling lostAll`.

Reply:
+1